src/databricks/labs/ucx/aws/credentials.py

```diff
@@ -181,3 +181,46 @@ def run(self, prompts: Prompts, include_names: set[str] | None = None) -> list[C
         else:
             logger.info("No IAM Role migrated to UC Storage credentials")
         return execution_result
+
+
+class IamRoleCreation:
+
+    def __init__(
+        self,
+        installation: Installation,
+        ws: WorkspaceClient,
+        resource_permissions: AWSResourcePermissions,
+    ):
+        self._output_file = "aws_iam_role_creation_result.csv"
+        self._installation = installation
+        self._ws = ws
+        self._resource_permissions = resource_permissions
+
+    @staticmethod
+    def _print_action_plan(iam_list: list[AWSUCRoleCandidate]):
+        # print action plan to console for customer to review.
+        for iam in iam_list:
+            logger.info(f"Role:{iam.role_name} Policy:{iam.policy_name} Paths:{iam.resource_paths}")
+
+    def save(self, migration_results: list[CredentialValidationResult]) -> str:
+        return self._installation.save(migration_results, filename=self._output_file)
+
+    def run(
+        self, prompts: Prompts, *, single_role=True, role_name="UC_ROLE", policy_name="UC_POLICY"
+    ) -> list[CredentialValidationResult]:
+
+        iam_list = self._resource_permissions.uc_roles_list(
+            single_role=single_role, role_name=role_name, policy_name=policy_name
+        )
+        if not iam_list:
+            logger.info("No IAM Role created")
+            return []
+        self._print_action_plan(iam_list)
+        plan_confirmed = prompts.confirm(
+            "Above UC Compatible IAM roles will be created and granted access to the corresponding paths, "
+            "please review and confirm."
+        )
+        if plan_confirmed is not True:
+            return []
+
+        return self._resource_permissions.create_uc_roles(iam_list)
```

Comment:
this function is called from the cli, so should not return anything

Reply:
OK